### DISCUSSION TITLE
annot.get_text() for highlight annotations?

### DISCUSSION THREAD

**Discussion by djoltes** [pymupdf/PyMuPDF#2846] [Q&A] [answered]
I see in the current doc that get_text() for annotations only works at present for certain types: "This currently only delivers relevant data for annotation types ‘FreeText’ and ‘Stamp’. Other types return an empty string (or equivalent objects)."

Are there plans to extend this to include retrieving text for highlight annotations? 

In the meantime, it appears I can grab the rectangle and use Page.get_textbox(rect) instead.

**Comment by JorjMcKie** [comment, marked as answer]
No, there are no such plans.
In contrast to text annotations, text marker annots contain no text themselves. They can very well highlight and image or an empty area without complaining. An this is no error condition either - completely legal.

**Reply by djoltes** [reply]
Makes sense. The query was more for planning purposes; I'll just use the rect to retrieve any text. Thanks!
